save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt

```diff
@@ -19,19 +22,28 @@ class TestService(private val configProperties: ConfigProperties) {
     @Autowired
     private lateinit var testRepository: TestRepository
 
+    @Autowired
+    private lateinit var testSuiteRepository: TestSuiteRepository
+
     /**
      * @param tests
      */
-    fun saveTests(tests: List<Test>) {
-        testRepository.saveAll(tests)
+    fun saveTests(tests: List<TestDto>) {
+        tests.map { testDto ->
+            testSuiteRepository.findById(testDto.testSuiteId).ifPresent {
+                testRepository.findByHash(testDto.hash).ifPresentOrElse(
```

Comment:
This should be the outer lambda; if you don't need to make a query to `testSuiteRepository` if the test is present, why make it unconditionally?